exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -102,6 +125,29 @@ public void setOption(OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, final OptionType type) {
+    try {
+      SystemOptionManager.getValidator(name); // ensure the option exists
+    } catch (final IllegalArgumentException e) {
+      throw UserException.validationError(e)
+        .build(logger);
```

Comment:
The function call does not have to be within the context of a query e.g. unit tests can use ```SystemOptionManager.getValidator(...)```. But callers, within the context of a query, will wrap the exception in a UserException.